engine/namegen.mli

```diff
@@ -125,3 +125,12 @@ val rename_bound_vars_as_displayed :
 val compute_displayed_name_in_gen :
   (evar_map -> int -> 'a -> bool) ->
   evar_map -> Id.Set.t -> Name.t -> 'a -> Name.t * Id.Set.t
+
+(** [universe_binders_with_opt_names ref l]
+
+    If [l] is [Some univs] return the universe binders naming the bound levels
+    of [ref] by [univs] (skipping Anonymous). May error if the lengths mismatch.
+
+    Otherwise return the bound universe names registered for [ref]. *)
```

Comment:
Maybe you could add a warning to indicate that the function is inefficient and not to be used at runtime?

Reply:
Even greater could be to place this function above `engine/tactics`, if possible of course.

Reply:
>Even greater could be to place this function above engine/tactics, if possible of course.

Sure it fits well in printing/printer IMO.